# USED FOR SETTINGS SOME SOLVER OPTIONS
horizon:

  urdf_relative_path: "description/urdf/awesome_leg_complete_sliding_all.urdf" # urdf relative path (wrt to the package)
 
  media_relative_path: "media/horizon_trot/sliding_all" # relative path where graphs, a copy of the opt results etc.. are saved

  simulation_name: "slower_trot5" # (opt results and graphs are automatically saved inside package_path/media/task_name/sub_task_name/today_date/simulation_name) 
  # to differentiate between simulations of the same task_type, sub_task_type carried out on the same date, modify the simulation_name before each simulation

  opt_results_rel_path: "opt_results/horizon_trot/sliding_all" # auxiliary folder used for storing the LAST solution 
  
  joint_cntrl:

    hip_joint:

      control_mode: "position" # "position", "velocity", "effort"
      stiffness: 1000.0
      damping: 5.0

    knee_joint:

      control_mode: "position"
      stiffness: 2000.0
      damping: 5.0

  horizon_solver:

    save_sol_as_init: false # if true, the solution is also saved as a candidate for future optimization initializations

    employ_opt_init: false # whether to use the initial guess or not
    
    variable_dt:

      problem_settings:

        n_nodes: 100 # total number of nodes
        n_takeoff: 50 # node @ takeoff
        dt: 0.01 # delta t between nodes of the optimizer

        forward_vel: 1 # forware velocity of the trot

        tip_ground_clearance: 0.1 # enforced tip clearance at a specified point in the flight phase (specified by "flight_phase_tip_clearance_percentage")
        flight_phase_tip_clearance_percentage: 0.6 # 0.0-1.0; when the tip clearance constraint is to be enforced w.r.t. the flight phase (e.g. 0.5-> @ 50%)
 
        solver:
          name: "ipopt"
          tolerance: 0.0001
          max_iter: 1000
          linear_solver_name: "ma57"    

        test_rig:
          lb: -0.4
          ub:  0.4      

        cost_weights:
          contact_force: 0.001
          weight_fictitious_actuation: 10.0
          small_q_p_ddot: 0.1
          weight_hip_i_d: 1.0
          weight_knee_i_d: 1.0
          weight_forward_vel: 100000.0
          
